dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/download/DownloadStateStore.java

```diff
@@ -74,30 +76,47 @@ public boolean hasStarted(File stateDir, String objectId) {
 
   public ObjectSpecification getProgress(File stateDir, String objectId) throws IOException {
     log.debug("Loading local progress for {} from {}", objectId, stateDir.toString());
-    ObjectSpecification spec = loadSpecification(stateDir, objectId);
+    val spec = loadSpecification(stateDir, objectId);
     log.debug("Completed loading local object specification (meta file)");
-    for (Part part : spec.getParts()) {
+    for (val part : spec.getParts()) {
       log.debug("Checking md5 for part {}", part.getPartNumber());
       if (isCompleted(stateDir, objectId, part)) {
-        Part completedPart = loadPart(stateDir, objectId, getPartName(part));
+        val completedPart = loadPart(stateDir, objectId, getPartName(part));
         // Copy download md5 into ObjectSpecification
         part.setMd5(completedPart.getMd5());
+      } else {
+        // part is not complete - check if it has expired
+        if (PresignedUrlValidator.isUrlExpired(part.getUrl())) {
+          val expiredMsg =
+              String
+                  .format("Presigned URL's have expired because download was not completed in alloted period. Restarting.");
+          val ise = new IllegalStateException(expiredMsg);
+          throw new NotRetryableException(ise);
+        }
       }
     }
     return spec;
   }
 
+  /**
+   * A marker file is written to the working directory after each part is downloaded and the MD5 checksum is verified.
+   * Checking to see if a part was completed previously is based on the presence of this marker file.
+   * @param stateDir - working directory for download
+   * @param objectId - object id
+   * @param part - metadata about part - really just for part number
+   * @return
+   */
   private boolean isCompleted(File stateDir, String objectId, Part part) {
-    File partFile = new File(getObjectStateDir(stateDir, objectId), getPartName(part));
+    val partFile = new File(getObjectStateDir(stateDir, objectId), getPartName(part));
     return partFile.exists();
   }
 
   public void commit(File stateDir, String objectId, Part part) {
     log.debug("Attempting to commit {} part {} to {}", objectId, part.getPartNumber(), stateDir.toString());
-    ObjectMapper mapper = new ObjectMapper();
+    val mapper = new ObjectMapper();
```

Comment:
Relatively expensive to create object mappers. Use `Jackson.DEFAULT` instead, or create static field.